src/graphql/thegraph/FeeTierDistributionQuery.ts

```diff
@@ -42,28 +37,27 @@ export default function useFeeTierDistributionQuery(
   token0: string | undefined,
   token1: string | undefined,
   interval: number
-) {
-  const [data, setData] = useState<FeeTierDistributionQuery$data | null>(null)
-  const [error, setError] = useState<any>(null)
-  const [isLoading, setIsLoading] = useState(true)
-  const chainId = useAppSelector((state) => state.application.chainId)
+): { error: ApolloError | undefined; isLoading: boolean; data: FeeTierDistributionQuery } {
+  const {
+    data,
+    loading: isLoading,
+    error,
+  } = useQuery(query, {
+    variables: {
+      token0: token0?.toLowerCase(),
+      token1: token1?.toLowerCase(),
+    },
+    pollInterval: interval,
+    client: apolloClient,
+    fetchPolicy: 'cache-and-network',
+  })
 
-  const refreshData = useCallback(() => {
-    if (token0 && token1 && chainId) {
-      fetchQuery<FeeTierDistributionQueryType>(environment, query, {
-        token0: token0.toLowerCase(),
-        token1: token1.toLowerCase(),
-      }).subscribe({
-        next: setData,
-        error: setError,
-        complete: () => setIsLoading(false),
-      })
-    }
-  }, [token0, token1, chainId])
-
-  // Trigger fetch on first load
-  useEffect(refreshData, [refreshData, token0, token1])
-
-  useInterval(refreshData, interval, true)
-  return { error, isLoading, data }
+  return useMemo(
```

Comment:
nit. don't think you need to memo this one since `data`, `loading` and `error` should be stable from apollo

Reply:
It has come about to make the standard practice for all our graphql queries to memoize the return data so while it might be stable I do like keeping with the standard personally